src/Microsoft.ML.SearchSpace/Option/ChoiceOption.cs

```diff
@@ -26,7 +26,7 @@ public ChoiceOption(params object[] choices)
             var distinctChoices = choices.Distinct();
             Contracts.Check(distinctChoices.Count() == choices.Length, "choices must not contain repeated values");
 
-            Choices = distinctChoices.OrderBy(x => x).ToArray();
+            Choices = distinctChoices.Select(o => Parameter.FromObject(o)).ToArray();
```

Comment:
>Select [](http://example.com/codeflow?start=38&length=6)

Do we care optimizing the allocations in such places? I am asking because `Select()` will create intermediate object and then we'll create the array from that result.

Reply:
Not quite care, as in most of the cases the candidate value for `choice` will not exceed 10, and this constructor only gets called once during training (when creating `choice` option). So the major cost will still be the training cost.